include/splay.h

```diff
@@ -263,11 +348,43 @@ template <class Visitor>
 void
 SplayNode<V>::visit(Visitor &visitor) const
 {
-    if (left)
-        left->visit(visitor);
-    visitor(data);
-    if (right)
-        right->visit(visitor);
+    // in-order walk through tree using Morris Traversal
+    SplayNode<V> *cur = const_cast<SplayNode<V> *>(this);
```

Comment:
visit and walk are indeed nearly identical. The only call to a "walk()"-function I've found in the source at all (except for in test-suite/splay.cc) was in src/String.cc and I can't see where it comes from. Squid builds without the walk-method, so it may be superfluous.

Reply:
String.cc code is inside `DEBUGSTRINGS` guards AFAICT. I hope that code, if enabled, (does not build in master and) can be removed together with walk(). I have not checked that yet.

Reply:
> String.cc code is inside `DEBUGSTRINGS` guards AFAICT. I hope that code, if enabled, (does not build in master and) can be removed together with walk(). I have not checked that yet.

I checked. Fortunately, that code does not build since 2013 commit 5082718 removed StringRegistry class declaration. I will post a PR to remove DEBUGSTRINGS code.

Reply:
>  I will post a PR to remove DEBUGSTRINGS code.

Done: #1436. I will see what I can do about the tests.

Reply:
> I will see what I can do about the tests.

I proposed to remove Splay::walk() from the tests and Splay itself in #1437. Landing that PR will address this change request.

Reply:
> Landing that PR will address this change request.

Splay::walk() is officially gone as of master/v7 commit 06ad882. I am resolving this change request.